joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala

```diff
@@ -314,9 +314,23 @@ trait KtPsiToAst {
     val parameters = withIndex(ktFn.getValueParameters.asScala.toSeq) { (p, idx) =>
       astForParameter(p, idx)
     }.flatMap(_.root.collectAll[NewMethodParameterIn])
-    val bodyAst = ktFn.getBodyBlockExpression match {
-      case blockExpr if blockExpr != null => astForBlock(blockExpr, None)
-      case _                              => Ast(NewBlock())
+
+    val bodyBlockAstMaybe =
+      ktFn.getBodyBlockExpression match {
+        case blockExpr if blockExpr != null => Some(astForBlock(blockExpr, None))
+        case _                              => None
+      }
+
+    val bodyAst = bodyBlockAstMaybe match {
+      case Some(ast) => ast
+      case None =>
+        val bodyBlock = blockNode("", "")
+        ktFn.getBodyExpression match {
+          case expr if expr != null =>
+            blockAst(bodyBlock, astsForExpression(expr, None).toList)
+          case _ =>
+            blockAst(bodyBlock, List())
+        }
     }
```

Comment:
This could be one match. E.g., you could wrap `ktFn.getBodyBlockExpression` like `Option(ktFn.getBodyBlockExpression)` which will automatically result in `None` if its `null`.

Reply:
yess, much nicer, done